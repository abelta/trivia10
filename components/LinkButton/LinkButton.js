import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-native';
import { Button } from '@shoutem/ui';

const LinkButton = ({ children, history, style, styleName, to }) => {
  const redirect = () => history.push(to);
  return (
    <Button onPress={redirect} style={style} styleName={styleName}>
      {children}
    </Button>
  );
};

LinkButton.propTypes = {
  children: PropTypes.node.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  history: PropTypes.object.isRequired,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
  styleName: PropTypes.string,
  to: PropTypes.string.isRequired,
};

LinkButton.defaultProps = {
  style: null,
  styleName: '',
};

export default withRouter(LinkButton);
