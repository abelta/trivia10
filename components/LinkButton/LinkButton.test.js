import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-native';
import { Text } from 'react-native';
import LinkButton from '.';

describe('LinkButton', () => {
  const tree = renderer.create(
    <Router>
      <LinkButton to="/">
        <Text>text</Text>
      </LinkButton>
    </Router>,
  ).toJSON();

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
