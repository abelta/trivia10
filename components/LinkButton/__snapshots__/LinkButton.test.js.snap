// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#ffffff",
      "borderRadius": 2,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingLeft": 15,
      "paddingRight": 5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    text
  </Text>
</View>
`;
